zuul-core/src/main/java/com/netflix/zuul/netty/filter/ZuulFilterChainHandler.java

```diff
@@ -121,15 +122,18 @@ private void sendResponse(final StatusCategory statusCategory, final int status,
             headers.add("Content-Length", "0");
             zuulResponse.finishBufferedBodyIfIncomplete();
             responseFilterChain.filter(zuulResponse);
-            fireEndpointFinish(true);
+            fireEndpointFinish(true, ctx);
```

Comment:
One slight nuance here with the change to `fireEndpointFinish`, since we close out the body on L123 (`zuulResponse.finishBufferedBodyIfIncomplete()`) and fire into the response filter chain, I don't think the new `finishResponseFilters` function will have an issue as no filter should be waiting for a body? Just double checking 😅

Reply:
True, but in the case of that static response, it won't pop into the if block that checks if the filter chain is waiting for a buffered body.